.upstream-tests/test/CMakeLists.txt

```diff
@@ -42,6 +42,18 @@ set(LIBCUDACXX_TEST_COMPILER_FLAGS
   -I${CMAKE_SOURCE_DIR}/include"
   CACHE INTERNAL "Flags for libcxx testing." FORCE)
 
+if (${CMAKE_CUDA_COMPILER_ID} STREQUAL "Clang")
+  set(LIBCUDACXX_TEST_COMPILER_FLAGS
+    "${LIBCUDACXX_TEST_COMPILER_FLAGS} \
+    -Xclang -fcuda-allow-variadic-functions \
+    -Xclang -Wno-unused-parameter \
+    -Wno-unknown-cuda-version")
+
+  set(LIBCUDACXX_TEST_LINKER_FLAGS
+    "${LIBCUDACXX_TEST_LINKER_FLAGS} \
+    -L${CUDA_TOOLKIT_ROOT_DIR}/lib64 -lcuda -lcudart")
```

Comment:
You should use:
```cmake
find_package(CUDAToolkit)
target_link_libraries( tests PRIVATE CUDA::cuda_driver CUDA::cudart)
```

Reply:
You're a saint Robert.

I thought this was done for me when enabling CUDA as a language. That's not the case?

Reply:
If CMake is generating the build files it will make sure that cudart is part of the link line ( with controls over static / shared ). It seems that in your case you are passing this information down to lit which does the build generation so you might need to stick with the 'bad' ( original ) approach.